src/tests/JIT/HardwareIntrinsics/Arm/Shared/Helpers.cs

```diff
@@ -5218,6 +5218,33 @@ public static float MultiplyExtended(float op1, float op2)
             }
         }
 
+        public static float TrigonometricMultiplyAddCoefficient(float op1, float op2, byte imm)
+        {
+            int index = (op2 < 0) ? (imm + 8) : imm;
```

Comment:
have we verified the behavior of this API with invalid values? If it is undefined, are we making the test robust to handle it?

Reply:
Like in #104681, the tests seem to work fine with out-of-bound values, though I can easily modify the tests to skip validation for such values.

Reply:
I liked the way you have combined the table of `sin` and `cos` in a single lookup.

Reply:
> Like in #104681, the tests seem to work fine with out-of-bound values, though I can easily modify the tests to skip validation for such values.

That will be great.

Reply:
> I liked the way you have combined the table of sin and cos in a single lookup.

Thanks, I wish I could say that was my idea, but the ARM docs suggested it